### PR TITLE
[SYCLomatic] Add some cuSPARSE/cuFFT/cuSolver examples for query-api-mapping

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

File: clang/examples/DPCT/lit_cuFFT.cu [hunk 1]
```diff
@@ -0,0 +1,99 @@
+// RUN: dpct --cuda-include-path="%cuda-path/include" --query-api-mapping=cufftExecZ2Z | FileCheck %s -check-prefix=cufftExecZ2Z
```
Comment (wenhuiNi) [inline clang/examples/DPCT/lit_cuFFT.cu]: maybe we need remove this file?

**Review by wenhuiNi — CHANGES_REQUESTED** [review]

File: clang/examples/DPCT/lit_cuFFT.cu (hunk 1 above)
Comment (zhiweij1) [reply clang/examples/DPCT/lit_cuFFT.cu]: removed

**Review by tangjj11 — APPROVED** [review]
LGTM

File: clang/test/dpct/query_api_mapping/cuSPARSE/lit_cuSPARSE.cu [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+// cusparseSpMM_preprocess: CUDA API:
+// cusparseSpMM_preprocess-NEXT:   cusparseSpMM_preprocess(
+// cusparseSpMM_preprocess-NEXT:       handle /*cusparseHandle_t*/, transa /*cusparseOperation_t*/,
+// cusparseSpMM_preprocess-NEXT:       transb /*cusparseOperation_t*/, alpha /*const void **/,
+// cusparseSpMM_preprocess-NEXT:       a /*cusparseConstSpMatDescr_t*/, b /*cusparseConstDnMatDescr_t*/,
+// cusparseSpMM_preprocess-NEXT:       beta /*const void **/, c /*cusparseDnMatDescr_t*/,
+// cusparseSpMM_preprocess-NEXT:       computetype /*cudaDataType*/, algo /*cusparseSpMMAlg_t*/,
+// cusparseSpMM_preprocess-NEXT:       workspace /*void **/);
+// cusparseSpMM_preprocess-NEXT: Is migrated to:
+// cusparseSpMM_preprocess-EMPTY:
```
Comment (zhimingwang36) [inline clang/test/dpct/query_api_mapping/cuSPARSE/lit_cuSPARSE.cu:14]: migration result is empty?

Comment (zhiweij1) [reply clang/test/dpct/query_api_mapping/cuSPARSE/lit_cuSPARSE.cu:14]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**